agent/src/com/jonnyzzz/teamcity/virtual/VMDetector.java

```diff
@@ -104,7 +129,7 @@ private String executeCommandWithShell(@NotNull final String name, @NotNull fina
           return null;
         }
 
-        return result.getStdout();
+        return result.getOutLines();
```

Comment:
The trick behind `ExecResult#getOutLines` is that it splits the `ExecResult#getStdout` by newlines. This code may yield empty entries on the array or white-space only entries.

Let's return the whole output here (as it was before the change) and process it closer to each of commands. This will allow to avoid any changes around vagrant detection at all